src/fides/api/models/pre_approval_webhook.py

```diff
@@ -24,13 +33,26 @@ def persist_obj(
 
 
 class PreApprovalWebhookReply(Base):
+    """
+    Stores the callback response to the PreApprovalWebhook to determine if the Privacy
+    Request is eligible to be automatically approved
+    """
+
     webhook_id = Column(
-        String, ForeignKey(PreApprovalWebhook.id_field_path), nullable=False
+        String,
+        ForeignKey(PreApprovalWebhook.id_field_path, ondelete="SET NULL"), index=True
     )
     privacy_request_id = Column(
         String, ForeignKey("privacyrequest.id", ondelete="SET NULL"), index=True
     )  # Which privacy request this webhook response belongs to
     is_eligible = Column(Boolean, nullable=False)
+
     privacy_request = relationship(
         "PrivacyRequest", back_populates="pre_approval_webhook_replies", uselist=False  # type: ignore
     )
+
+    __table_args__ = (
+        UniqueConstraint(
+            "webhook_id", "privacy_request_id", name="webhook_id_privacy_request_id"
+        ),
+    )
```

Comment:
This concept was already in the database but needed to be added to the model to match.